processor/src/main/java/apoc/processor/ExtensionClassWriter.java

```diff
@@ -25,16 +29,37 @@ public void write(List<String> procedureSignatures,
                       List<String> userFunctionSignatures) {
 
         try {
-            JavaFile.builder("apoc", defineClass(procedureSignatures, userFunctionSignatures))
+            String path = getProjectPath();
+            final TypeSpec typeSpec = defineClass(procedureSignatures, userFunctionSignatures, path);
+
+            JavaFile.builder("apoc", typeSpec)
                     .build()
                     .writeTo(filer);
         } catch (IOException e) {
             throw new RuntimeException(e);
         }
     }
 
-    private TypeSpec defineClass(List<String> procedureSignatures, List<String> userFunctionSignatures) {
-        return TypeSpec.classBuilder("ApocSignatures")
+    private String getProjectPath() throws IOException {
+        // create and delete a file to retrieve the current project (`core` or `extended`)
+        FileObject resource = filer.createResource(StandardLocation.SOURCE_OUTPUT, "", "tmp", (Element[]) null);
+        String projectPath = resource.getName();
+        resource.delete();
```

Comment:
This feel a bit over complicated to me. Cannot we not e.g. send in a boolean to the write method depending on if we call it from core or extended?

Reply:
Changed with a boolean

Reply:
I was more thinking about if we could avoid creating a temporary file, but looking closer I think it might not be so easy to do it in a smarter way. With your change it is at least a bit cleaner :)